articles/tips-and-tricks-for-deep-learning/index.md

```diff
@@ -0,0 +1,252 @@
+
+We could initialize the weights randomly. But will this work? Yes, it usually works just fine, but it doesn't guarantee an absolute asymmetry.
+
+There are two main commonly used ways that we could initialize the weights in our neural network. These include the use of Transfer Learning and the Xavier Initializer. 
+
+Let's discuss these two.
+
+##### Transfer Learning
+
+Transfer learning is a deep learning technique whereby a model is developed for one task but then re-used as a starting point for a separate task. This technique helps in leveraging the pre-trained weight of that model in our model. Besides, transfer learning saves a lot of time as training a deep learning model from scratch requires a lot of training data and computing power. 
+
+This technique can be used in various ways, depending on the amount of data one has. These include leveraging transfer learning with small, medium, and large training sizes. 
+
+Let's take a look at pictorial examples of how this is done.
+
+![Transfer Learning with a small training size](/engineering-education/tips-and-tricks-for-deep-learning/small-tl.PNG)<br>
+
+*[Image Source: StanFord](https://stanford.edu/~shervine/teaching/cs-230/cheatsheet-deep-learning-tips-and-tricks)* 
+
+Here, all layers in the neural network with pre-trained weights are frozen. It is only the softmax layer whose weights can be trained on. This is ideal if you have a small training size.
+
+![Transfer Learning with a medium training size](/engineering-education/tips-and-tricks-for-deep-learning/medium-tl.PNG)<br>
+
+*[Image Source: StanFord](https://stanford.edu/~shervine/teaching/cs-230/cheatsheet-deep-learning-tips-and-tricks)*
+
+If you have a medium training size, perform transfer learning on a neural network where most of the layers with the pre-trained weights, excluding the last layer and the softmax layer, have been frozen as shown above. This gives you a little more room to train your data on. 
+
+![Transfer Learning with a large training size](/engineering-education/tips-and-tricks-for-deep-learning/large-tl.PNG)<br>
+
+*[Image Source: StanFord](https://stanford.edu/~shervine/teaching/cs-230/cheatsheet-deep-learning-tips-and-tricks)*
+
+In cases where you have a large training data, you can leverage training weights on almost all the layers and softmax layer while initializing your weights on the pre-trained data. 
+
+Transfer Learning works well in two fields of Artificial Intelligence:
+1. Computer Vision
+2. Natural Language Processing (NLP)
+
+##### Xavier Initialization
+
+Xavier initialization is also known as the Glorot Initialization. It is a weight initialization technique used to initialize the weights in a deep neural network. This technique helps avoid neuron activation functions, starting in very saturated and dead regions.
```

Comment:
Define what saturated and dead regions are. It is necessary to explain how Xavier's initialization solves a problem, and why most of the time, we just use random initialization.